x-pack/plugins/infra/public/apps/discover_app.tsx

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { interpret } from 'xstate';
+import type { DiscoverStart } from '@kbn/discover-plugin/public';
+import type { InfraClientStartDeps, InfraClientStartExports } from '../types';
+import type { LogViewColumnConfiguration, ResolvedLogView } from '../../common/log_views';
+import { createLogViewStateMachine, DEFAULT_LOG_VIEW } from '../observability_logs/log_view_state';
+import { MESSAGE_FIELD, TIMESTAMP_FIELD } from '../../common/constants';
+
+export const renderApp = (plugins: InfraClientStartDeps, pluginStart: InfraClientStartExports) => {
+  const { discover } = plugins;
+  const { logViews } = pluginStart;
+
+  const machine = createLogViewStateMachine({
+    initialContext: { logViewReference: DEFAULT_LOG_VIEW },
+    logViews: logViews.client,
+  });
+
+  const service = interpret(machine)
+    .onTransition((state) => {
+      if (
+        state.matches('checkingStatus') ||
+        state.matches('resolvedPersistedLogView') ||
+        state.matches('resolvedInlineLogView')
+      ) {
+        return redirectToDiscover(discover, state.context.resolvedLogView);
+      } else if (
+        state.matches('loadingFailed') ||
+        state.matches('resolutionFailed') ||
+        state.matches('checkingStatusFailed')
+      ) {
+        return redirectToDiscover(discover);
+      }
+    })
+    .start();
+
+  return () => {
+    // Stop machine interpreter after navigation
+    service.stop();
+  };
+};
+
+const redirectToDiscover = (discover: DiscoverStart, resolvedLogView?: ResolvedLogView) => {
+  if (!resolvedLogView) {
+    return discover.locator?.navigate({});
+  }
+
+  const columns = parseColumns(resolvedLogView.columns);
+  const {
+    allowNoIndex,
+    id,
+    name,
+    namespaces,
+    sourceFilters,
+    timeFieldName,
+    type,
+    typeMeta,
+    version,
+  } = resolvedLogView.dataViewReference;
```

Comment:
Interestingly, the `title` doesn't seem to get set correctly. Maybe we missed that when we create the ad-hoc data view during log view resolution. This is the first time we're actually exposing the data view to an external app that displays the title, so we didn't notice in the past. Do we want to try to fix this in the log view resolution function and pass on the `title` here?

Reply:
On second thought, would it make sense to use the `toSpec()` method of the data view or is there a reason to selectively pass only some propterties?

Reply:
+1  on the usage of `toSpec()` method, I didn't think about it but it works perfectly for this case.

About setting the title, I see it gets set to the index pattern specified or the related data view pattern, is that wrong? What should the title be in case of an ad-hoc data view for example?

Reply:
Very nice, thank you!